### PR TITLE
Correct version check in UI

### LINKED ISSUE #5405: Cannot update netdata
##### Bug report summary
I can't update netdata with
`bash <(curl -Ss https://my-netdata.io/kickstart.sh --no-updates)`
Error: 
`
curl: option --no-updates: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
`
Also, when at the end of the netdata's installation I got this error:
`
rm: cannot remove './netdata-updater.sh': No such file or directory
`

##### OS / Environment
Kernel : Linux 4.4.0-140-generic x86_64
OS : Ubuntu 16.04

##### Netdata version (ouput of `netdata -V`)
netdata v1.12.0-22-nightly

##### Component Name
?

##### Steps To Reproduce
Try to update netdata

##### Expected behavior
Update correctly

### PR DESCRIPTION
**Pull request by cakrit**
##### Summary
Fixes #5405

##### Component Name
daemon, web

##### Additional Information
Support stable and nightly release channel.
Use github releases instead of latest versions, get nightlies from GCS

It's WIP because it can't be merged until we get GCS configured to permit cross-origin requests, but otherwise it's ok. @paulfantom please check the version comparison logic, the old one didn't work any more. I think that this one is fine, it has passed all the tests I did.

### CODE REVIEW

File: web/gui/main.js [hunk 1]
```diff
@@ -2832,39 +2834,95 @@ function versionsMatch(v1, v2) {
...
+        // Split patch: format could be e.g. 0-22-nightly
+        s1=s1[2].split('-');
+        s2=s2[2].split('-');
+
+        n1 = parseInt(s1[0],10);
+        n2 = parseInt(s2[0],10);
+        if ( n1 < n2 ) return false;
+        else if ( n1 > n2 ) return true;
+
+        n1 = (s1.length > 1)?parseInt(s1[1],10):0;
```
Comment (gmosx) [inline web/gui/main.js]: Minor formatting note, pls add some spacing around `?` and `:`.

**Comment by gmosx** [comment]
> It's WIP because it can't be merged until we get GCS configured to permit cross-origin requests, 
> but otherwise it's ok.

Who is handling this?

**Comment by cakrit** [comment]
> Who is handling this? 

I resolved it, don't worry about it. The settings in Google don't affect it at all:

> Note: CORS configuration only affects requests to XML API endpoints. JSON API endpoints allow CORS requests, regardless of CORS settings on the target bucket. Requests to the endpoint storage.cloud.google.com do not allow CORS requests.

We were trying it with storage.cloud.google.com, I used the JSON API at the end.

File: web/gui/main.js (hunk 1 above)
Comment (cakrit) [reply web/gui/main.js]: Fixed it

File: web/api/formatters/charts2json.c [hunk 2]
```diff
@@ -4,6 +4,41 @@
...
+    static int use_stable = -1;
+
+    if (use_stable == -1) {
+		char filename[FILENAME_MAX + 1];
+        snprintfz(filename, FILENAME_MAX, "%s/.environment", netdata_configured_user_config_dir);
+        procfile *ff = procfile_open(filename, "=", PROCFILE_FLAG_DEFAULT);
+        if(!ff) {
+            use_stable=1;
+        } else {
+            ff = procfile_readall(ff);
```
Comment (ktsaou) [inline web/api/formatters/charts2json.c]: Add here:

```c
procfile_set_quotes(ff, "'\"");
```

Then then `value` will always be unquoted.

Comment (cakrit) [reply web/api/formatters/charts2json.c]: Done, thanks.

**Pull request merged**